kernel/common/binding.rb

```diff
@@ -54,4 +54,12 @@ def eval(expr, filename=nil, lineno=nil)
 
     Kernel.eval(expr, self, filename, lineno)
   end
+
+  def local_variables
+    eval('local_variables')
+  end
+
+  def receiver
+    @self
```

Comment:
@jemc I agree. The ```attr_accessor :self``` could be replaced with regular
```def self=(recv) ...``` and ```def self ...``` methods refering to ```@receiver```, which would make the purpose of the ivar clearer, without breaking client code depending on ```self```.